language/vm/src/deserializer.rs

```diff
@@ -1107,13 +1205,41 @@ fn load_field_instantiations(
 }
 
 /// Deserializes a `FunctionDefinition`.
-fn load_function_def(cursor: &mut Cursor<&[u8]>) -> BinaryLoaderResult<FunctionDefinition> {
+fn load_function_def(cursor: &mut VersionedCursor) -> BinaryLoaderResult<FunctionDefinition> {
     let function = load_function_handle_index(cursor)?;
 
-    let flags = read_u8(cursor).map_err(|_| {
+    let flags = read_u8(cursor.cursor()).map_err(|_| {
         PartialVMError::new(StatusCode::MALFORMED).with_message("Unexpected EOF".to_string())
     })?;
-    let is_public = (flags & FunctionDefinition::PUBLIC) != 0;
+
+    let visibility = match cursor.version() {
+        VERSION_1 => {
```

Comment:
Probably fine to just use 1 and 2 here. I don't see the constant doing much :)

Reply:
To me having a constant imported from somewhere feels a bit safer... But yeah, purely personal preference.

Reply:
I'm not sure how much safer it is if VERSION_1 is always just 1?

Maybe it conveys a bit more intent? I could see that maybe, but I think `version() < 2` has been fairly clear in my usecase

Reply:
Yeah, I think VERSION_1 is probably going to be 1 all the time, and same for VERSION_2. Maybe the usecase I was thinking about is, say, I want to look at the places where there is a special handling of VERSION_n, I could do a `grep VERSION_n` to pinpoint the places.

Reply:
Yeah thats not a bad one. Let's do the constants then